### ISSUE TITLE
Toy Regression: Only one iteration of SGD is done in training net one time

### ISSUE THREAD

**Issue by ghost** [facebookarchive/caffe2#1498]
![image](https://user-images.githubusercontent.com/5510968/33049363-5df3721e-ce25-11e7-8700-10b3a23d6d6c.png)
![image](https://user-images.githubusercontent.com/5510968/33049382-70eef1d6-ce25-11e7-8292-463639174afc.png)
NOTE: In the above codes, X and Y_gt are initialized in init_net. So in each training, we get constant input X and Y_gt.
NOTE: I ran train_net two times, and got different w and b.

I thought caffe2 would get the best w and b with SGD in one training( workspace.RunNet(train_net.Proto().name) ), i.e. many iterations of gradient descent will be done to get the best w and b for loss function in one training. Looks like I was wrong:(

So my question is:  only one iteration of Gradient  Descent is done in one training, if we want to get the best w and b, we need to run the train_net for many times with the same X and Y_gt.
Is that right?

Thanks!

**Comment by pietern** [comment]
You are right, one call to RunNet only does a single iteration.

To make W and B converge you have to call RunNet more iterations.
